src/config/reader.rs

```diff
@@ -28,7 +26,7 @@ impl ConfigReader {
     }
     Ok(config)
   }
-  async fn from_file_path(file_path: &str) -> anyhow::Result<Config> {
+  pub(crate) async fn from_file_path(file_path: &str) -> anyhow::Result<Config> {
```

Comment:
```suggestion
  pub async fn from_file_path(file_path: &str) -> anyhow::Result<Config> {
```

Probably `pub(crate)` makes more sense, but to be consistent we can keep it `pub`